e2etests/test/configurations.js

```diff
@@ -13,6 +13,13 @@ const getErrorDetailString = require('./testUtils').getErrorDetailString;
 const hubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
 const registry = Registry.fromConnectionString(hubConnectionString);
 
+
+function waitForEventualConsistency(callback) {
+  // Some of these tests occasionally fails, presumably because the registry doesn't update quickly
+  // enough. Sleep for an arbitrary 2 seconds to account for this.
+  setTimeout(() => callback(), 2000);
```

Comment:
Do we have a guarantee that it's eventually consistent by 2 seconds, or is this just some number you chose that seemed to work?

Reply:
2 seconds is totally arbitrary.